### ISSUE TITLE
Question about TabM model

### ISSUE THREAD

**Issue by brokim-ensol** [qile2000/LAMDA-TALENT#17]
Thank you for your efforts.

With regard to TabM model, I found that there are some deviations from original repository(https://github.com/yandex-research/tabm) especially about class name, method name and etc.

And futhermore, I would like to test the "tabm-mini with piecewise-linear encoding/embeddings" architecture as mentioned in the original paper. In the pager, it is mentioned that the performance of "tabm-mini with piecewise-linear encoding/embeddings" architecture is better than the vanillar version of "tabm". However in this repository 'bins' attribute is commented (https://github.com/qile2000/LAMDA-TALENT/blob/658e9481f827d685818ba89203910b5037c310d6/LAMDA_TALENT/model/models/tabm.py#L30) whereas 'bins' attribute still exists (https://github.com/yandex-research/tabm/blob/663a4922f750186b8071cdd2bebf4214422c2c5a/bin/model.py#L91)

I'm wondering what is the latest version of "tabm" model. Is it yours?? or not revised yet in this repository?

Thank you for your understanding.

**Comment by brokim-ensol** [comment]
Oh, I see. piecewise-linear encoding/embedding is enabled by default setting when loading "tabm" model. https://github.com/qile2000/LAMDA-TALENT/blob/658e9481f827d685818ba89203910b5037c310d6/LAMDA_TALENT/model/utils.py#L602

**Comment by 6sy666** [comment]
> Thank you for your efforts.
> 
> With regard to TabM model, I found that there are some deviations from original repository(https://github.com/yandex-research/tabm) especially about class name, method name and etc.
> 
> And futhermore, I would like to test the "tabm-mini with piecewise-linear encoding/embeddings" architecture as mentioned in the original paper. In the pager, it is mentioned that the performance of "tabm-mini with piecewise-linear encoding/embeddings" architecture is better than the vanillar version of "tabm". However in this repository 'bins' attribute is commented (
> 
> [LAMDA-TALENT/LAMDA_TALENT/model/models/tabm.py](https://github.com/qile2000/LAMDA-TALENT/blob/658e9481f827d685818ba89203910b5037c310d6/LAMDA_TALENT/model/models/tabm.py#L30)
> 
> Line 30 in [658e948](/qile2000/LAMDA-TALENT/commit/658e9481f827d685818ba89203910b5037c310d6)
> 
>  # bins: None | list[Tensor], 
> ) whereas 'bins' attribute still exists (https://github.com/yandex-research/tabm/blob/663a4922f750186b8071cdd2bebf4214422c2c5a/bin/model.py#L91)
> I'm wondering what is the latest version of "tabm" model. Is it yours?? or not revised yet in this repository?
> 
> Thank you for your understanding.


Thank you for your detailed feedback and for pointing out the differences between our implementation and the original repository.

In our implementation of the TabM model, we have chosen to use the PLR (Periodic activation, Linear, ReLU) module by default instead of the PLE (Piecewise Linear Encoding) embedding used in the original paper and repository. This decision was made to facilitate easier comparison with other methods in our toolkit.

If you would like to use the "tabm-mini" architecture as mentioned in the original paper, you can make the following changes:

1. **Change the model type in the configuration file:**
   Modify the `model_type` field in the configuration file from `"tabm"` to `"tabm-mini"` here:
   [tabm.json#L4](https://github.com/qile2000/LAMDA-TALENT/blob/658e9481f827d685818ba89203910b5037c310d6/LAMDA_TALENT/configs/default/tabm.json#L4)

2. **Update the model initialization in the code:**
   If you are tuning hyperparameters, you will also need to change the model type in the code here:
   [utils.py#L608](https://github.com/qile2000/LAMDA-TALENT/blob/658e9481f827d685818ba89203910b5037c310d6/LAMDA_TALENT/model/utils.py#L608)

Our implementation is a modified version of the original TabM model, tailored to fit within our toolkit's framework and to support the PLR module. We appreciate your understanding and hope this clarifies the differences.

By the way, we have recently added a new `use_float` option to our toolkit. If you want to train the model using `float` data type for faster training speed, you can run the training script with the `--use_float` flag, like this:

```bash
python train_model_deep.py --use_float ...
```

In our original implementation, the toolkit used `double` data type by default for higher precision. However, switching to `float` can speed up training while maintaining reasonable accuracy for most use cases. 

If you have any further questions or need additional assistance, please feel free to reach out.

**Comment by brokim-ensol** [comment]
I really appreciate your kind explanation.
